### PR TITLE
Model support for character creation

### PR DESCRIPTION
**Pull request by Vassyli**
Adds support for model creation.

### CODE REVIEW

File: src/Models/Character.php [hunk 1]
```diff
@@ -11,18 +12,34 @@
...
     /** @Id @Column(type="integer") @GeneratedValue */
     private $id;
     /** @Column(type="string", length=50, unique=true); */
     private $name;
     /** @Column(type="text", unique=true); */
     private $displayName;
     /** @Column(type="integer", options={"default" = 10}) */
-    private $health = 10;
-    /** @Column(type="integer", options={"default" = 10}) */
     private $maxhealth = 10;
```
Comment (austenmc) [inline src/Models/Character.php:24]: Let's use consistent capitalization: `$maxHealth`

File: src/Models/Character.php [hunk 2]
```diff
@@ -46,4 +63,14 @@ public function getName(): string {
...
     }
+    
+    /**
+     * Sets the maximum health of a character to a given value. It also sets the
+     * health if none has been set yet.
+     * @param int $maxhealth
+     */
+    protected function setMaxhealth(int $maxhealth) {
+        $this->maxhealth = $maxhealth;
+        $this->health = $this->health??$this->maxhealth;
```
Comment (austenmc) [inline src/Models/Character.php:74]: spacing: `??`

File: src/Tools/Model/Creator.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        $entity = new self();
+        
+        foreach(self::$fillable as $field) {
+            if(isset($arguments[$field])) {
+                $methodname = "set".$field;
+                $value = $arguments[$field];
+                
+                $entity->$methodname($value);
+            }
+        }
```
Comment (austenmc) [inline src/Tools/Model/Creator.php:41]: You should assert or at least log when passed arguments that aren't fillable.

File: src/Models/Character.php [hunk 4]
```diff
@@ -46,4 +63,14 @@ public function getName(): string {
     protected function generateDisplayName() {
         $this->displayName = $this->name;
     }
+    
+    /**
+     * Sets the maximum health of a character to a given value. It also sets the
+     * health if none has been set yet.
+     * @param int $maxhealth
+     */
+    protected function setMaxhealth(int $maxhealth) {
```
Comment (austenmc) [inline src/Models/Character.php:72]: `setMaxHealth`

File: src/Models/Character.php (hunk 2 above)
Comment (austenmc) [inline src/Models/Character.php:74]: Honestly, I dont think this should do anything w/ health. If I call a mutator on a class w/ an instance variable of the same name, I dont expect it to mutate anything except that instance variable. Seems like a surprising side effect.

File: src/Exceptions/WrongTypeException.php [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace LotGD\Core\Exceptions;
+
+/**
+ * Exception if a variable has the wrong type.
+ */
+class WrongTypeException extends CoreException {
+}
```
Comment (austenmc) [inline src/Exceptions/WrongTypeException.php:9]: nit: consistent spacing, either have an extra newline (like above) or don't.

**Comment by austenmc** [comment]
Made some comments. Feel free to fix up and commit.

File: src/Models/Character.php (hunk 2 above)
Comment (Vassyli) [inline src/Models/Character.php:74]: It does only have an effect on $health if $health is NULL which is important during character creation - A character should be created at full health.

Comment (Vassyli) [inline src/Models/Character.php:74]: It is in principle just another way do define a default value, it's not that surprising.

On the other hand, if it's not given during character::create, $health stays null.

**Pull request closed without merging**